heartbeat/monitors/active/http/config.go

```diff
@@ -153,3 +156,27 @@ func (c *compressionConfig) Validate() error {
 
 	return nil
 }
+
+// Validate validates of the Config object is valid or not
+func (c *Config) Validate() error {
+	if len(c.Hosts) == 0 && len(c.URLs) == 0 {
+		return fmt.Errorf("hosts is a mandatory parameter")
+	}
+
+	if len(c.URLs) != 0 {
+		c.Hosts = append(c.Hosts, c.URLs...)
+	}
+
+	for i := 0; i < len(c.Hosts); i ++ {
+		host := c.Hosts[i]
+		if _, err := url.ParseRequestURI(host); err != nil {
```

Comment:
This could use a comment like:

> For consistency we allow users to enter both URLs and host:port pairs in the `hosts` field, since internally HTTP checks must operate on hosts. Rather than parsing these using separate parsers, we simply see if the host is a valid URL, and if not, see if it becomes valid if http(s) is added in front of it.